### PR TITLE
IGAPP-40: Build config documentation

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by klinzo** [comment]
I figured an documentation directly in the code would be better in this case as you really have to change it if you change behavior. Would you rather like a separate documentation?

File: build-configs/BuildConfigType.js [hunk 1]
```diff
@@ -3,61 +3,70 @@
 import type { ThemeType } from './ThemeType'
 import type { TranslationsType } from 'translations'
 
+/**
```
Comment (maxammann) [inline build-configs/BuildConfigType.js:6]:
```suggestion
/*
```

As this is not a jsdoc comment

File: build-configs/BuildConfigType.js [hunk 2]
```diff
@@ -3,61 +3,70 @@
...
+ * Build Configs
+ * These are the flow types of our build configs and therefore define the structure and available options.
```
Comment (maxammann) [inline build-configs/BuildConfigType.js:8]: I think we should explain here the difference between build options and feature flags

File: build-configs/BuildConfigType.js [hunk 3]
```diff
@@ -3,61 +3,70 @@
...
 |}
 
 export type FeatureFlagsType = {|
   ...SelectedCityType,
-  pois: boolean,
-  newsStream: boolean,
-  pushNotifications: boolean,
-  sentry: boolean,
-  developerFriendly: boolean
+  pois: boolean, // Enables POIs and maps, can be disabled by cities.
```
Comment (maxammann) [inline build-configs/BuildConfigType.js:24]:
```suggestion
  pois: boolean, // Enables POIs and maps. It is also possible to disable pois via our API for each city individually.
```

File: build-configs/BuildConfigType.js [hunk 4]
```diff
@@ -3,61 +3,70 @@
...
+// Available on all platforms
 export type CommonBuildConfigType = {|
   appName: string,
   appIcon: string,
   cmsUrl: string,
-  switchCmsUrl?: string,
-  shareBaseUrl: string,
-  allowedHostNames: Array<string>,
-  internalLinksHijackPattern: string,
+  switchCmsUrl?: string, // Secondary api url to use, selectable by clicking ten times on the location marker (native).
```
Comment (maxammann) [inline build-configs/BuildConfigType.js:36]:
```suggestion
  switchCmsUrl?: string, // Secondary api url to use, selectable by clicking ten times on the location marker (only works for native right now).
```

**Review by maxammann — APPROVED** [review]
If you want you can use JSDoc instead of plain comments. Apart from that this documentation should be in code :+1: 

https://jsdoc.app/tags-param.html

**Review by Taggotty — APPROVED** [review]

File: build-configs/README.md [hunk 5]
```diff
@@ -7,24 +7,38 @@ We use the concept of build configs to build different versions of the app. A bu
...
+* [Structure and Available Options](#structure-and-available-options)
+* [Technical Information](#technical-information)
+* [Whitelabelling](#whitelabelling)
+
 ## Available Build Configs
 
 * **integreat**: Build config for [Integreat](https://integreat-app.de).
-* **malte**: Build config for the [Malte](https://www.malteser-werke.de/malte-app.html).
+* **malte**: Build config for [Malte](https://www.malteser-werke.de/malte-app.html).
+* **aschaffenburg/abapp**: TODO
```
[the remark this answers is not in the document]
Comment (maxammann) [reply build-configs/README.md:22]: aschaffenburg!

**Pull request merged**